src/expr/src/scalar/func/impls/timestamp.rs

```diff
@@ -121,19 +121,23 @@ impl fmt::Display for CastTimestampToTimestampTz {
 #[derive(
     Arbitrary, Ord, PartialOrd, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash, MzReflect,
 )]
-pub struct CastTimestampToTimestamp {
+pub struct AdjustTimestampPrecision {
     pub from: Option<TimestampPrecision>,
     pub to: Option<TimestampPrecision>,
 }
 
-impl<'a> EagerUnaryFunc<'a> for CastTimestampToTimestamp {
+impl<'a> EagerUnaryFunc<'a> for AdjustTimestampPrecision {
     type Input = CheckedTimestamp<NaiveDateTime>;
     type Output = Result<CheckedTimestamp<NaiveDateTime>, EvalError>;
 
     fn call(
         &self,
         a: CheckedTimestamp<NaiveDateTime>,
     ) -> Result<CheckedTimestamp<NaiveDateTime>, EvalError> {
+        // If the from and to precisions are same, then this is effectively a no-op.
+        // Adding a soft_assert to flag such instances.
+        mz_ore::soft_assert!(self.to != self.from);
```

Comment:
Why `soft_assert` over an early return? It doesn't seem incorrect to cast something to the same precision, just potentially wasteful.

Either way, can you make sure there's a test somewhere for this case for all the updated funcs.

Reply:
I see what you mean, we can return early. I will do that instead.